accession-commons-core/src/main/java/uk/ac/ebi/ampt2d/commons/accession/core/AccessioningService.java

```diff
@@ -27,18 +27,39 @@
 
 import java.util.List;
 
+/**
+ * Service that provides various accessioning features for objects
+ *
+ * @param <MODEL> Specifies the type of the Accession model
+ * @param <HASH> Hash key used for accessioning
+ * @param <ACCESSION> Accession ID of the object
+ */
 public interface AccessioningService<MODEL, HASH, ACCESSION> {
 
+    /**
+     * Get accessions for a list of messages. It looks for the object's accessions in a repository, and if they don't
+     * exist, generate new ones, storing them in the repository
+     *
+     * @param messages List of messages to be accessioned
+     * @return Wrapper containing the object that has been accessioned
+     * @throws AccessionCouldNotBeGeneratedException when accession could not be generated
+     */
     List<AccessionWrapper<MODEL, HASH, ACCESSION>> getOrCreate(List<? extends MODEL> messages)
             throws AccessionCouldNotBeGeneratedException;
 
+    /**
+     * Get accessions for a list of objects by generating Hash
```

Comment:
The fact that a hash is generated is an implementation detail and I think it should be dropped from the method documentation. Also other methods' documentation refer to 'find', so the final doc would look like this:

```suggestion
     * Find the accessions associated with a list of objects.
```